runtime/runtime_controller.h

```diff
@@ -41,53 +117,329 @@ class RuntimeController final : public WindowClient {
+  ///
+  /// @return     If the message was dispatched to the running root isolate.
+  ///             This may fail is an isolate is not running.
+  ///
   bool DispatchPlatformMessage(fml::RefPtr<PlatformMessage> message);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Dispatch the specified pointer data message to the running
+  ///             root isolate.
+  ///
+  /// @param[in]  packet  The pointer data message to dispatch to the isolate.
+  ///
+  /// @return     If the pointer data message was dispatched. This may fail is
+  ///             an isolate is not running.
+  ///
   bool DispatchPointerDataPacket(const PointerDataPacket& packet);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Dispatch the semantics action to the specified accessibility
+  ///             node.
+  ///
+  /// @param[in]  id      The identified of the accessibility node.
+  /// @param[in]  action  The semantics action to perform on the specified
+  ///                     accessibility node.
+  /// @param[in]  args    Optional data that applies to the specified action.
+  ///
+  /// @return     If the semantics action was dispatched. This may fail if an
+  ///             isolate is not running.
+  ///
   bool DispatchSemanticsAction(int32_t id,
                                SemanticsAction action,
                                std::vector<uint8_t> args);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Gets the main port identifier of the root isolate.
+  ///
+  /// @return     The main port identifier. If no root isolate is running,
+  ///             returns `ILLEGAL_PORT`.
+  ///
   Dart_Port GetMainPort();
 
+  //----------------------------------------------------------------------------
+  /// @brief      Gets the debug name of the root isolate. But default, the
+  ///             debug name of the isolate is derived from its advisory script
+  ///             URI, advisory main entrypoint and its main port name. For
+  ///             example, "main.dart$main-1234" where the script URI is
+  ///             "main.dart", the entrypoint is "main" and the port name
+  ///             "1234". Once launched, the isolate may re-christen itself
+  ///             using a name it selects via `setIsolateDebugName` in
+  ///             `window.dart`. This name is purely advisory and only used by
+  ///             instrumentation and reporting purposes.
+  ///
+  /// @return     The debug name of the root isolate.
+  ///
   std::string GetIsolateName();
 
+  //----------------------------------------------------------------------------
+  /// @brief      Returns if the root isolate has any live receive ports.
+  ///
+  /// @return     True if there are live receive ports, False otherwise. Return
+  ///             False is the root isolate is not running as well.
```

Comment:
Would be helpful to state why this matters here - e.g. this effectively tells you if the isolate is running or not, right?

Reply:
A running isolate can have no receive ports.... right?